services/rest/src/test/java/mil/nga/giat/geowave/service/rest/RestServerTest.java

```diff
@@ -1,5 +1,289 @@
+
 public class RestServerTest
 {
+	@Rule
+	public TemporaryFolder tempFolder = new TemporaryFolder();
+
+	@BeforeClass
+	public static void runServer() {
+		// Ensure the config directory exists.
+		File configDirectory = ConfigOptions.getDefaultPropertyPath();
+		if (!configDirectory.exists() && !configDirectory.mkdir()) throw new RuntimeException(
+				"Unable to create directory " + configDirectory);
+		RestServer.main(new String[] {});
+
+	}
+
+	// Tests geowave/config/set and list
+	@Test
+	public void geowave_config_set_list()
+			throws ResourceException,
+			IOException,
+			ParseException {
+
+		File configFile = tempFolder.newFile("test_config");
+
+		// create a new store named "store1", with type "memory"
+		ClientResource resourceAdd = new ClientResource(
+				"http://localhost:5152/geowave/config/addstore");
+		resourceAdd.setChallengeResponse(
+				ChallengeScheme.HTTP_BASIC,
+				"admin",
+				"password");
+		Form formAdd = new Form();
+		formAdd.add(
+				"name",
+				"store1");
+		formAdd.add(
+				"storetype",
+				"memory");
+		formAdd.add(
+				"default",
+				"false");
+		formAdd.add(
+				"config_file",
+				configFile.getAbsolutePath());
+		resourceAdd.post(
+				formAdd).write(
+				System.out);
+
+		// set key=store1, value=store2
+		ClientResource resourceSet = new ClientResource(
+				"http://localhost:5152/geowave/config/set");
+		resourceSet.setChallengeResponse(
+				ChallengeScheme.HTTP_BASIC,
+				"admin",
+				"password");
+		Form formSet = new Form();
+		formSet.add(
+				"key",
+				"store1");
+		formSet.add(
+				"value",
+				"store2");
+		formSet.add(
+				"config_file",
+				configFile.getAbsolutePath());
+		resourceSet.post(
+				formSet).write(
+				System.out);
+
+		// testing list
+		ClientResource resourceList = new ClientResource(
+				"http://localhost:5152/geowave/config/list");
+		resourceList.setChallengeResponse(
+				ChallengeScheme.HTTP_BASIC,
+				"admin",
+				"password");
+		resourceList.addQueryParameter(
+				"config_file",
+				configFile.getAbsolutePath());
+		String text = resourceList.get(
+				MediaType.APPLICATION_JSON).getText();
+
+		JSONParser parser = new JSONParser();
+		JSONObject obj = (JSONObject) parser.parse(text);
+		String name = (String) obj.get("store1");
+		assertTrue(
+				"'name' is 'value'",
+				name.equals("store2"));
+
+		// remove the store named "store1"
+		ClientResource resourceRm = new ClientResource(
+				"http://localhost:5152/geowave/config/rmstore");
+		resourceRm.setChallengeResponse(
+				ChallengeScheme.HTTP_BASIC,
+				"admin",
+				"password");
+		Form formRm = new Form();
+		formRm.add(
+				"name",
+				"store1");
+		formRm.add(
+				"config_file",
+				configFile.getAbsolutePath());
+		resourceRm.post(
+				formRm).write(
+				System.out);
+	}
+
+	// Tests geowave/config/addstore, cpstore, rmstore
+	@Test
+	public void geowave_config_store()
+			throws ResourceException,
+			IOException {
+
+		File configFile = tempFolder.newFile("test_config");
```

Comment:
TemporaryFolder guarantees this will be a new file: http://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html#newFile(java.lang.String)